modules/blockbuilder/blockbuilder.go

```diff
@@ -75,6 +75,12 @@ var (
 		Name:      "fetch_errors_total",
 		Help:      "Total number of errors while fetching by the consumer.",
 	}, []string{"partition"})
+	metricOwnedPartitions = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "tempo",
+		Subsystem: "block_builder",
+		Name:      "owned_partitions",
+		Help:      "Indicates partition ownership by this block-builder (1 = owned).",
+	}, []string{"partition"})
```

Comment:
Adding the partition state would be very nice.

Reply:
Good idea